### PR TITLE
Adding building for source info

### PR DESCRIPTION
**Pull request by sishida**
Signed-off-by: Sarah Ishida <sishida@us.ibm.com>

PR for https://github.com/eclipse/codewind/issues/350

### CODE REVIEW

**Comment by tetchel** [comment]
It looks like all this does is move the Developing section further up. I think it should definitely be kept below all the user documentation. What is that motivation for this change?

**Comment by tetchel** [comment]
I think it should be Contributing > Developing > Building from source

**Comment by sishida** [comment]
@tetchel I moved the sections so they're Developing > Building from source, but I left Contributing at the bottom because the other repos have Contributing at the bottom.

**Comment by tetchel** [comment]
it makes sense to read the contributing section before the developing section; potential developers will want to start there, and the set of users who will be interested in contributing vs developing is definitely larger

**Comment by sishida** [comment]
OK. I guess most other repos don't have a Developing section like how this one does. I will go ahead and move it up.

**Comment by tetchel** [comment]
one more change; 'the codewind repository' (or just the 'codewind' part) should link to https://github.com/eclipse/codewind

**Comment by sishida** [comment]
@tetchel Good call--since we've moved the content to the VS Code repo, it'd be good to include a link to the Codewind repo. I have added it.

File: README.md [hunk 1]
```diff
@@ -50,3 +49,9 @@ Submit issues and contributions:
...
 - The extension bundles dependency executables. These are gitignored, but should be kept up-to-date on your local system with the same versions used in the `Jenkinsfile` `parameters` section. Run `dev/bin/pull.sh` to download the dependencies. Also see `dev/bin/README.txt`.
 - The [`prebuild`](https://github.com/eclipse/codewind-vscode/blob/master/dev/prebuild.js) script is used in the CI builds to build separate versions of the extension for VS Code and Theia, since each of those has some commands that the other does not. It deletes inapplicable commands from the `package.json`, and does not modify any ts/js code. Run this before `vsce package` to get a closer-to-production build, but be ready to revert the changes.
+
+## Building Codewind from the source
+1. Clone the [`codewind`](https://github.com/eclipse/codewind) repository.
+2. Clone the `codewind-vscode` repo.
+3. Run the `./script/build.sh` script to run the Codewind build, or run the `./run.sh` script to build and start Codewind.
```
Comment (tetchel) [inline README.md]: one more clarifying detail would be to 'run `codewind/script/build.sh`' or `codewind/script/run.sh`

so they know to look under codewind, not under codewind-vscode, for those scripts

File: README.md [hunk 2]
```diff
@@ -50,3 +49,9 @@ Submit issues and contributions:
...
+4. Run the plug-in by following the instructions in **Developing**.
```
Comment (tetchel) [inline README.md]: 'extension', not 'plug-in'

Comment (tetchel) [reply README.md]: let's just say "Run the VS Code extension" for maximum clarity

File: README.md [hunk 3]
```diff
@@ -50,3 +49,9 @@ Submit issues and contributions:
...
 - The extension bundles dependency executables. These are gitignored, but should be kept up-to-date on your local system with the same versions used in the `Jenkinsfile` `parameters` section. Run `dev/bin/pull.sh` to download the dependencies. Also see `dev/bin/README.txt`.
 - The [`prebuild`](https://github.com/eclipse/codewind-vscode/blob/master/dev/prebuild.js) script is used in the CI builds to build separate versions of the extension for VS Code and Theia, since each of those has some commands that the other does not. It deletes inapplicable commands from the `package.json`, and does not modify any ts/js code. Run this before `vsce package` to get a closer-to-production build, but be ready to revert the changes.
+
+## Building Codewind from the source
+1. Clone the [`codewind`](https://github.com/eclipse/codewind) repository.
+2. Clone the `codewind-vscode` repo.
+3. Run the `codewind/script/build.sh` script to run the Codewind build, or run the `codewind/script/run.sh` script to build and start Codewind.
```
Comment (tetchel) [inline README.md]: sorry, my previous comment was misleading, I didn't realize our run/build scripts have to be run from their directory or they will fail.

trying again, we should add a new step for `cd` to the codewind directory.
```
3. `cd` to the `script` directory of your codewind clone, `cd codewind/script`
3. Run `./build.sh` to run the Codewind build, or run `./run.sh` to build and start Codewind.
```

I would say either "run `./build.sh`" or "run the `build.sh` script", not "run the `./build.sh` script".

**Pull request merged**